ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -413,31 +421,27 @@ void NGraphEncapsulateOp::ComputeUsingParallelExecutor(OpKernelContext* ctx) {
   NGRAPH_VLOG(2) << "[PREFETCH] COMPUTE: Input: " << ctx->num_inputs()
                  << " Values: " << tf_input_tensors[0].DebugString();
   int step_id = ctx->step_id();
-  ngraph::Event event_compile("Compile", "", "");
+  ngraph::Event event_get_ng_item("GetExecutableAndTensors", "", "");
+  shared_ptr<PipelinedTensorsStore> pipelined_tensor_store;
+  std::tuple<int, PipelinedTensorVector, PipelinedTensorVector> io_tensors;
 
-  // Get ngraph executable and inputs information
+  // Get ngraph executable and inputs information and Pipelined tensors
   bool cache_hit;
-  OP_REQUIRES_OK(ctx, m_parallel_executor->GetNgExecutable(tf_input_tensors,
-                                                           ng_exec, cache_hit));
-
+  std::string serialized_ng_function;
+  OP_REQUIRES_OK(ctx, m_parallel_executor->GetExecutableFunctionAndTensors(
+                          tf_input_tensors, ng_exec, serialized_ng_function,
+                          pipelined_tensor_store, cache_hit));
+  io_tensors = pipelined_tensor_store->get_tensors();
+  OP_REQUIRES(ctx, !(std::get<0>(io_tensors) < 0),
+              errors::Internal("No free tensor available"));
```

Comment:
This is an important behaviour. Do we have a test for this? A test would be:
start running a graph with 3 threads, the first 2 threads will work, the last one is expected to throw an error.
so the end-to-end test would expect an error out of one of the 3 threads.

Reply:
Please file a ticket for this